# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/WasmBCE.cpp
# Commit: 349728105c40
# Full Hash: 349728105c40993896170f4c2f43e152a3466706
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-01-08 09:40:24
# Regressor Bug: 1642412
# File Overlap Count: 19
# Description:
#   Bug 1748602 - Backout bug 1642412. r=lth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D135346
# ==============================================================================

diff -r 32873022d2ed -r 349728105c40 js/src/jit/WasmBCE.cpp
--- a/js/src/jit/WasmBCE.cpp	Fri Jan 07 20:35:15 2022 +0000
+++ b/js/src/jit/WasmBCE.cpp	Fri Jan 07 20:43:16 2022 +0000
@@ -45,13 +45,7 @@
           MWasmBoundsCheck* bc = def->toWasmBoundsCheck();
           MDefinition* addr = bc->index();
 
-          // We only support bounds check elimination on wasm memory, not
-          // tables. See bug 1625891.
-          if (!bc->isMemory()) {
-            continue;
-          }
-
-          // Eliminate constant-address memory bounds checks to addresses below
+          // Eliminate constant-address bounds checks to addresses below
           // the heap minimum.
           //
           // The payload of the MConstant will be Double if the constant